aea/configurations/base.py

```diff
@@ -745,6 +746,13 @@ def __init__(
         self.dependencies = dependencies if dependencies is not None else {}
         self.description = description
 
+        if protocol_specification_id is not None:
+            self.protocol_specification_id = PublicId.from_str(
+                protocol_specification_id
+            )
```

Comment:
Are we sure we want to allow for this?

Reply:
for the compatibility with old packages

Reply:
check for None and auto set default value will stay till  transition of packages is done